src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+     * The era and other fields in {@link ChronoField} are defined by the chronology.
+     *
+     * @return the International fixed chronology, not null
+     */
+    @Override
+    public InternationalFixedChronology getChronology() {
+        return InternationalFixedChronology.INSTANCE;
+    }
+
+    /**
+     * Gets the era applicable at this date.
+     * <p/>
+     * The International fixed calendar system only has one era, 'CE',
+     * defined by {@link InternationalFixedEra}.
+     *
+     * @return the era applicable at this date, not null
+     */
+    @Override
+    public InternationalFixedEra getEra() {
+        return InternationalFixedEra.CE;
+    }
+
+    /**
+     * Returns the length of the month represented by this date.
+     * <p/>
+     * This returns the length of the month in days.
+     * Month lengths do not match those of the ISO calendar system.
+     *
+     * Since Leap Day / Year Day are not part of any month, their 'imaginary' month is of length 1.
+     *
+     * @return the length of the month in days: 28
+     */
+    @Override
+    public int lengthOfMonth() {
+        return month > 0 ? InternationalFixedChronology.DAYS_IN_MONTH : 1;
+    }
+
+    /**
+     * Returns the length of the year represented by this date.
+     * <p/>
+     * This returns the length of the year in days.
+     * Year lengths match those of the ISO calendar system.
+     *
+     * @return the length of the year in days: 365 or 366
+     */
+    @Override
+    public int lengthOfYear() {
+        return InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear() ? 1 : 0);
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalAdjuster adjuster) {
+        return (InternationalFixedDate) adjuster.adjustInto(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate with(final TemporalField field, final long newValue) {
+        if (field instanceof ChronoField) {
+            ChronoField f = (ChronoField) field;
+            getChronology().range(f).checkValidValue(newValue, f);
+
+            if (f == ChronoField.DAY_OF_MONTH || f == ChronoField.MONTH_OF_YEAR) {
+                if (newValue == 0) {
+                    return createYearDay(getProlepticYear());
+                }
+
+                if (newValue == -1) {
+                    return createLeapDay(getProlepticYear());
+                }
+            }
+
+            int dom = isYearDay() ? 21 : (getCalculatedDayOfMonth() / 7) * 7;
+            int d = day < 1 ? 1 : day % 7;
+            int nval = (int) newValue;
+
+            switch (f) {
+                case DAY_OF_WEEK:
+                case ALIGNED_DAY_OF_WEEK_IN_MONTH:
+                case ALIGNED_DAY_OF_WEEK_IN_YEAR:
+                    if (newValue == 0) {
```

Comment:
Check the test data comments, where I explain this a little more.  Essentially, 0 should only be a valid value for staying on (not moving) leap day or year day for week-related values (can't move to year day, because staying on leap day would be confusing).  It doesn't move leap day/year day because it's using those values they return.  But it's different than what's returned for any other day, so staying on them doesn't make sense.  Normal weeks don't have a 0 day of week, either, so it's not a valid value for them.